demos/python_demos/forward_tacotron_wavernn_demo/models/forward_tacotron_ie.py

```diff
@@ -0,0 +1,168 @@
+            for j in range(tot_duration.shape[1]):
+                pos1 = tot_duration[i, j]
+                index[i, pos:pos1, :] = j
+                pos = pos1
+            index[i, pos:, :] = j
+        return index
+
+    @staticmethod
+    def gather(a, dim, index):
+        expanded_index = [index if dim==i else np.arange(a.shape[i]).reshape([-1 if i==j else 1 for j in range(a.ndim)]) for i in range(a.ndim)]
+        return a[tuple(expanded_index)]
+
+    def load_network(self, model_xml):
+        model_bin_name = ".".join(osp.basename(model_xml).split('.')[:-1]) + ".bin"
+        model_bin = osp.join(osp.dirname(model_xml), model_bin_name)
+        print("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+        net = self.ie.read_network(model=model_xml, weights=model_bin)
+        return net
+
+    def create_exec_network(self, net):
+        exec_net = self.ie.load_network(network=net, device_name=self.device)
+        return exec_net
+
+    def infer_duration(self, sequence, alpha=1.0, non_empty_symbols=None):
+        out = self.duration_predictor_exec.infer(inputs={"input_seq": sequence})
+        duration = out["duration"] * alpha
+
+        duration = (duration + 0.5).astype('int').flatten()
+        duration = np.expand_dims(duration, axis=0)
+        preprocessed_embeddings = out["embeddings"]
+
+        if non_empty_symbols is not None:
+            duration = duration[:, :non_empty_symbols]
+            preprocessed_embeddings = preprocessed_embeddings[:,:non_empty_symbols]
+        indexes = self.build_index(duration, preprocessed_embeddings)
+        if self.verbose:
+            print("Index: {0}, duration: {1}, embeddings: {2}, non_empty_symbols: {3}"
+                  .format(indexes.shape, duration.shape, preprocessed_embeddings.shape, non_empty_symbols))
+
+        return self.gather(preprocessed_embeddings, 1, indexes)
+
+    def infer_mel(self, aligned_emb):
+        out = self.forward_exec.infer(inputs={"data": aligned_emb})
+        return out['mel']
+
+    def forward(self, text, alpha=1.0):
+        sequence = self.seq_to_indexes(text)
+        if len(sequence) <= self.duration_len:
+            non_empty_symbols = None
+            if len(sequence) < self.duration_len:
+                non_empty_symbols = len(sequence)
+                sequence += [_symbol_to_id[' ']] * (self.duration_len - len(sequence))
+            sequence = np.array(sequence)
+            sequence = np.expand_dims(sequence, axis=0)
+            if self.verbose:
+                print("Seq shape: {0}".format(sequence.shape))
+            aligned_emb = self.infer_duration(sequence, alpha, non_empty_symbols=non_empty_symbols)
+            if self.verbose:
+                print("AEmb shape: {0}".format(aligned_emb.shape))
+        else:
+            punctuation = '!\'(),.:;? '
+            delimiters = [_symbol_to_id[p] for p in punctuation]
+            # try to found optimal fragmentation for inference
```

Comment:
```suggestion
            # try to find optimal fragmentation for inference
```
Is it the same as
```py
                delimers ='.!?;:'
                for i, c in enumerate(line):
                    if (c in delimers and i - prev_begin > len_th) or i == len(line) - 1:
                        texts.append(line[prev_begin:i+1])
                        prev_begin = i
```
in `main()` with just a bit different set of delimiters?

Reply:
Yes, it is not a problem. The procedure should be little different for pipeline and network inference.

Reply:
Okay though it sounds strange, but I'm worried now that
```py
                time_s = time.perf_counter()
                mel = forward_tacotron.forward(text)
                time_e = time.perf_counter()
```
depends on the number of chunks text is split internally. This duration measurement may count different number if inferences from one call to another.

Reply:
You always sum them so it doesn't matter. Sorry.

Reply:
Yes. But in the end I just count summary time.